tests/test_docstring_options.py

```diff
@@ -40,3 +40,51 @@ def test_docstring_options():
     # Suppression of user-defined docstrings for non-function objects
     assert not m.DocstringTestFoo.__doc__
     assert not m.DocstringTestFoo.value_prop.__doc__
+
+    # Check overload configuration behaviour matches the documentation
+    assert m.test_overloaded4.__doc__ == (
+        "test_overloaded4(*args, **kwargs)\n"
+        "Overloaded function.\n"
+        "\n"
+        "1. test_overloaded4(arg0: int, arg1: int) -> int\n"
+        "\n"
+        "Add two integers together.\n"
+        "\n"
+        "2. test_overloaded4(arg0: float, arg1: float) -> float\n"
+        "\n"
+        "Add two floating point numbers together.\n"
+    )
+
+    assert m.test_overloaded5.__doc__ == (
+        "test_overloaded5(arg0: int, arg1: int) -> int\n"
+        "test_overloaded5(arg0: float, arg1: float) -> float\n"
+        "\n"
+        "Overloaded function.\n"
+        "\n"
+        "1. test_overloaded5(arg0: int, arg1: int) -> int\n"
+        "\n"
+        "Add two integers together.\n"
+        "\n"
+        "2. test_overloaded5(arg0: float, arg1: float) -> float\n"
+        "\n"
+        "Add two floating point numbers together.\n"
+    )
+
+    assert m.test_overloaded6.__doc__ == (
+        "test_overloaded6(arg0: int, arg1: int) -> int\n"
+        "test_overloaded6(arg0: float, arg1: float) -> float\n"
+        "test_overloaded6(arg0: None, arg1: None) -> None\n"
+        "\n"
+        "A function which adds two numbers.\n"
+        "\n"
+        "Internally, a simple addition is performed.\n"
+        "Both numbers can be None, and None will be returned."
+    )
+
+    assert m.test_overloaded7.__doc__ == (
+        "test_overloaded7(*args, **kwargs)\n"
+        "A function which adds two numbers.\n"
+        "\n"
```

Comment:
A complete nit, only affecting this test: I'd find it slightly easier to reach the "aha" point like this:
```
"A function which adds two numbers.\n\n"
```